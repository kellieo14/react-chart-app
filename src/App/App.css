:root {
    --black: #191c2e;
}

.App {
    background-color: var(--black);
    height: 100%;
    min-height: 100vh;
}

.App h1 {
    margin: 0;
    color: whitesmoke;
}

.chartTitle {
    text-transform: uppercase;
    color: lightgray;
    letter-spacing: 2px;
    font-size: 1em;
    margin-bottom: 25px;
}

.pie {
    margin-top: 50px;
}

.doughnut {
    margin-top: 50px;
}

.verticalBar {
    margin-top: 50px;
}

.horizontalBar {
    margin-top: 50px;
}

.densityRow {
    margin-top: 40px;
    padding: 10px;
}

.title-container h1 {
    position: absolute;
    margin: 0 auto;
    left: 0;
    right: 0;
    top: 40%;
    text-align: center;
    z-index: 1;
    text-shadow: 2px 2px 2px #912046, 0 0 1em #912046, 0 0 0.2em  #0080d6;
    color: var(--black);
}

.title-container  {
    font-family: 'Knewave', cursive;
    position: relative;
    display: inline-block;
    padding-top: 75px;
}

@media (max-width: 1000px) {
    .title-container h1 {
        font-size: 3em;
    }
}

@media (min-width: 1001px) {
    .title-container h1 {
        font-size: 5vw;
    }
}