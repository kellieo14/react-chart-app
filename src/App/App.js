import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CountriesPerContinent from "../components/CountriesPerContinent/CountriesPerContinent";
import PopulationPerContinent from '../components/PopulationPerContinent/PopulationPerContinent';
import PopulationPerCountry from '../components/PopulationPerCountry/PopulationPerCountry';
import LanguageByCountry from '../components/LanguageByCountry/LanguageByCountry';
import LanguageByPopulation from '../components/LanguageByPopulation/LanguageByPopulation';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import WorldMap from '../components/WorldMap/WorldMap';
import Density from '../components/Density/Density';
import { getCountries } from '../redux/actions/actions';

import './App.css';

const App = () => {
  const countries = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
  }, [dispatch]);

  return (
    <div className="App">
      <Container>
        <Row className='title-container'>
          <h1>World Data</h1>
          <WorldMap />
        </Row>
        <div className='charts'>
          <Row className='pie-row'>
            <Col lg={6}>
              <CountriesPerContinent countries={countries} />
            </Col>
            <Col lg={6}>
              <PopulationPerContinent countries={countries} />
            </Col>
          </Row>
          <Row className='chart-rows'>
            <Col lg={6}>
              <LanguageByPopulation countries={countries} />
            </Col>
            <Col lg={6}>
              <LanguageByCountry countries={countries} />
            </Col>
          </Row>
          <Row>
            <Col lg={6}>
              <PopulationPerCountry countries={countries} />
            </Col>
            <Col className='densityRow' lg={6}>
              <Density countries={countries} />
            </Col>
          </Row>
        </div>

      </Container>
    </div>
  );
}

export default App;
