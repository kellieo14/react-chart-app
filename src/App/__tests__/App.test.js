import React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App.js';
import apiData from '../../components/ApiData'

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../components/ApiData', () => ({
    __esModule: true,
    default: jest.fn()
}));

describe('App', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot', () => {
        const testRenderer = TestRenderer.create(<App />);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it('should call api', () => {
        shallow(<App />);
        expect(apiData).toHaveBeenCalledTimes(1);
    });
});