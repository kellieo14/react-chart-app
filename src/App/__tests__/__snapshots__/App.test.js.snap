// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="title-container row"
    >
      <h1>
        World Data
      </h1>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-12"
          >
            <img
              alt="world map"
              className="map"
              src="worldMap.png"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="charts"
    >
      <div
        className="pie-row row"
      >
        <div
          className="col-lg-6"
        >
          <div
            className=""
          >
            <div>
              <div
                className="doughnut"
              />
            </div>
          </div>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className=""
          >
            <div
              className="pie"
            >
              <h4
                className="chartTitle"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="chart-rows row"
      >
        <div
          className="col-lg-6"
        >
          <div>
            <div
              className="verticalBar"
            >
              <h4
                className="chartTitle"
              />
            </div>
          </div>
        </div>
        <div
          className="col-lg-6"
        >
          <div>
            <div
              className="horizontalBar"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-lg-6"
        >
          <div>
            <div
              className="verticalBar"
            >
              <h4
                className="chartTitle"
              />
            </div>
          </div>
        </div>
        <div
          className="densityRow col-lg-6"
        >
          <div>
            <div
              className="polar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
